runtime/ms_rest/lib/ms_rest/serialization.rb

```diff
@@ -391,7 +391,14 @@ def serialize_sequence_type(mapper, object, object_name)
       # @param model_name [String] Name of the model to retrieve.
       #
       def get_model(model_name)
-        Object.const_get(@context.class.to_s.split('::')[0...-1].join('::') + "::Models::#{model_name}")
+        consts = @context.class.to_s.split('::')
+        if consts.any?{ |const| const == 'Models' }
```

Comment:
Not so much a bug, but another way of including the serialization logic. Honestly, the serialization logic shouldn't need an anchor to the service class. 

JSONable will be decorated onto the model, rather than the @client, and as such, needs to understand the context so it can create a new version of the deserialized object.

Perhaps, this logic should be decorated onto the model and not known by the service class. I was thinking about doing this larger refactor, but didn't feel like it was my place to make sweeping changes like that.

Does this make sense?